### PR TITLE
Adds tests to number with unit + deletes unused code

### PR DESCRIPTION
**Pull request by ryan-bradford**
Signed-off-by: Ryan Bradford <rbradford@vmware.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] Tests for the changes have been added (for bug fixes / features)
-   [ ] Examples have been added / updated (for bug fixes / features)
-   [ ] Changelog has been updated

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [ ] Feature
-   [ ] Code style update (formatting, local variables)
-   [ ] Refactoring (no functional changes, no api changes)
-   [x] Build related changes
-   [ ] CI related changes
-   [ ] Documentation content changes
-   [ ] Example website changes
-   [ ] Version bump
-   [x] Other... Please describe: Testing!

## What does this change do?

1. Adds tests for number with unit form input. Also, there was a block of code in the `get displayName` that seemed to be un-callable and unused. I deleted this block.
2. VSCode (and potentially other editors) use the closest tsconfig to compile files. This means in the project `find-cypress-widget.ts` (which includes Cypress) was being added to the compilation of the project. This was causing errors to appear in the spec files. To get around this, I created a new tsconfig in just the /components directory that ecludes the find-cypress-widget.ts.

## What manual testing did you do?

1. Ran the examples app.
2. Navigated to the number with unit tab.
3. Changed the value to 4096 mb
4. Observed the display changing to 4096 mb
5. Set to readonly
6. Observed the correct display of 4096 mb
7. Change unit to GB
8. Set to 1 GB
9. Observe correct display of 1024 MB
10. Set to readonly
11. Observe correct display of 1024 MB

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

**Comment by juanmendes** [comment]
Please avoid testing done as 

> and made sure everything still worked.

Please list the steps you actually used to test.

File: package.json [hunk 1]
```diff
@@ -52,6 +52,7 @@
     "lit-html": "1.1.2",
     "messageformat": "1.0.2",
     "mousetrap": "1.6.5",
+    "parse5": "^6.0.1",
```
Comment (juanmendes) [inline package.json:55]: I don't see this being used

File: projects/components/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (juanmendes) [inline projects/components/tsconfig.json:1]: Why this change? Can you please explain in the PR description?

File: package.json (hunk 1 above)
Comment (ryan-bradford) [reply package.json:55]: it was because of some sudden build failures, will try and make it go away

File: projects/components/tsconfig.json (hunk 2 above)
Comment (ryan-bradford) [reply projects/components/tsconfig.json:1]: Sorry, forgot about this change. Adding to description and improving PR

**Review by irahov — APPROVED** [review]
